scripts/verify-structure.py

```
@@ -48,16 +58,144 @@
+        path_resolved = Path(normalized_str)
+
+        # Check if normalized path is under base directory
+        try:
+            path_resolved.relative_to(base_resolved)
+        except ValueError:
+            return None, f"{context}: Path escapes base directory: {relative_path}"
+
+        # Path is safe, return the path object for existence checks
+        # Use the original Path construction for the return value
+        full_path = base_dir / relative_path.lstrip("./")
+        return full_path, None
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)